classes/core/PKPString.php

```diff
@@ -399,23 +402,68 @@ public static function getAmbiguousExtensionsMap()
      * Strip unsafe HTML from the input text. Covers XSS attacks like scripts,
      * onclick(...) attributes, javascript: urls, and special characters.
      *
-     * @param string $input input string
-     * @param string $configKey The config section key['allowed_html', 'allowed_title_html']
+     * @param string|null   $input input string
+     * @param string        $key   The config section key['allowed_html', 'allowed_title_html']
      *
      * @return string
      */
-    public static function stripUnsafeHtml($input, $configKey = 'allowed_html')
+    public static function stripUnsafeHtml(?string $input, string $key = 'allowed_html'): string
     {
-        static $purifier;
-        if (!isset($purifier)) {
-            $config = HTMLPurifier_Config::createDefault();
-            $config->set('Core.Encoding', 'utf-8');
-            $config->set('HTML.Doctype', 'HTML 4.01 Transitional');
-            $config->set('HTML.Allowed', Config::getVar('security', $configKey));
-            $config->set('Cache.SerializerPath', 'cache');
-            $purifier = new HTMLPurifier($config);
+        if (!$input) {
+            return '';
         }
-        return $purifier->purify((string) $input);
+
+        static $sanitizer;
+        static $configKey;
+        static $allowedTagToAttributeMap;
+
+        if ($configKey !== $key) {
```

Comment:
This is meant to cache repeat calls with the same `$configKey`, right? We only have two different configurations for `$configKey` we use in practice at the moment; at the moment flipping back and forth between two values of `$configKey` will cause the parsing to be repeated. I'd suggest storing the parsed data in an array by `$configKey`.

Reply:
Moved both the attributes tag map and sanitizer as static cache as both needed at the end .